src/ampal/protein.py

```diff
@@ -555,26 +555,33 @@ def n_join(self, other, psi=-40.76, omega=-178.25, phi=-65.07,
         self.tags['assigned_ff'] = False
         return
 
-    def tag_sidechain_dihedrals(self, force=False, tag_rotamers=False):
-        """Tags each monomer with side-chain dihedral angles
+    def tag_sidechain_dihedrals(self, force=False):
+        """Tags each monomer with side-chain dihedral angles. For residues that
+        do not have any rotamers (Alanine and Glycine) the rotamer is tagged as
+        0 and chi_angles as None.
 
         force: bool, optional
             If `True` the tag will be run even if `Residues` are
             already tagged.
-        rotamers: bool, optional
-            If `True` it will tag rotamers as tags.
         """
-        tagged = ["chi_angles" in x.tags.keys() for x in self._monomers]
-        if tag_rotamers:
-            tagged = ["rotamers" in x.tags.keys() for x in self._monomers]
-        if (not all(tagged)) or force:
+        chi_tagged = ["chi_angles" in x.tags.keys() for x in self._monomers]
+        rot_tagged = ["rotamers" in x.tags.keys() for x in self._monomers]
+        if (not all(chi_tagged)) or (not all(rot_tagged)) or force:
             for monomer in self._monomers:
-                if tag_rotamers:
-                    chi_angles, rotamer = measure_sidechain_torsion_angles(monomer, verbose=False, return_rotamers=True)
+                chi_angles_rotamer = measure_sidechain_torsion_angles(monomer, verbose=False)
+                if chi_angles_rotamer:
+                    chi_angles, rotamer = chi_angles_rotamer
                     monomer.tags["rotamers"] = rotamer
+                    monomer.tags["chi_angles"] = chi_angles
+                # Rotamer not found: either residue does not have rotamers or
+                # it is not a common residue
                 else:
-                    chi_angles = measure_sidechain_torsion_angles(monomer, verbose=False)
-                monomer.tags["chi_angles"] = chi_angles
+                    if monomer.mol_letter == "G" or monomer.mol_letter == "A":
```

Comment:
You could move this test to before the `measure_sidechain_torsion_angles` call to save a bit of compute.

Reply:
Good catch